sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubClientFactory.cs

```diff
@@ -66,19 +66,24 @@ internal EventHubProducerClient GetEventHubProducerClient(string eventHubName, s
                                 RetryOptions = _options.ClientRetryOptions,
                                 ConnectionOptions = _options.ConnectionOptions
                             });
-                    }
-
-                    return new EventHubProducerClient(
-                        NormalizeConnectionString(info.ConnectionString, eventHubName),
-                        new EventHubProducerClientOptions
-                        {
-                            RetryOptions = _options.ClientRetryOptions,
-                            ConnectionOptions = _options.ConnectionOptions
-                        });
+                    });
+                }
+                else
+                {
+                    return _producerCache.GetOrAdd(NormalizeConnectionString(info.ConnectionString, eventHubName), key =>
+                    {
+                        return new EventHubProducerClient(
+                            NormalizeConnectionString(info.ConnectionString, eventHubName),
```

Comment:
Can we cache the value into local variable instead of computing it twice?

Reply:
Fixed